### PR TITLE
ROC-5265: Refactor admissions dashboard (full and part)

### PR DESCRIPTION
**Pull request by BenDoherty0103**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5265

### Change description

Refactoring of logic concerning full and part admit dashboard statuses

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [x] Task list and task completeness checks updated
- [ ] Check and send page updated
- [x] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/app/claims/models/claim-states/states-paid-states.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export enum StatesPaidStates {
+  STATES_PAID = 'states-paid'
```
Comment (andrewwa-kainos) [inline src/main/app/claims/models/claim-states/states-paid-states.ts]: States paid can be via full defence or part admit - is a single state enough?

Comment (BenDoherty0103) [reply src/main/app/claims/models/claim-states/states-paid-states.ts]: Full defence has states called FD_ALREADY PAID, FD_ALREADY_PAID_RESPONSE, , FD_ALREADY_PAID_ACCEPT, and FD_ALREADY_PAID_REJECT.

File: src/main/app/claims/models/claim-states/part-admission-states.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  PA_PAY_BY_SET_DATE_REJECTED_WITH_MEDIATION_CLAIMANT_WITH_MEDIATION = 'pa-pay-by-set-date-rejected-with-mediation-claimant-with-mediation',
+  PA_PAY_BY_SET_DATE_REJECTED_WITHOUT_MEDIATION = 'pa-pay-by-set-date-rejected-without-mediation',
+
+  PA_PAY_IN_INSTALMENTS_WITH_MEDIATION = 'pa-pay-in-instalments-with-mediation',
+  PA_PAY_IN_INSTALMENTS_WITHOUT_MEDIATION = 'pa-pay-in-instalments-without-mediation',
+  PA_PAY_IN_INSTALMENTS_REJECTED_WITH_MEDIATION_CLAIMANT_WITHOUT_MEDIATION = 'pa-pay-in-instalments-rejected-with-mediation-claimant-without-mediation',
+  PA_PAY_IN_INSTALMENTS_REJECTED_WITH_MEDIATION_CLAIMANT_WITH_MEDIATION = 'pa-pay-in-instalments-rejected-with-mediation-claimant-with-mediation',
+  PA_PAY_IN_INSTALMENTS_REJECTED_WITHOUT_MEDIATION = 'pa-pay-in-instalments-rejected-without-mediation',
+
+  PA_PAY_BY_SPECIFIED_DATE = 'pa-pay-by-specified-date',
```
Comment (robertparkinson) [inline src/main/app/claims/models/claim-states/part-admission-states.ts]: this doesn't seem to be used and I don't think it makes sense as you've covered everything above in the pay_by_set_datte

File: src/main/app/claims/models/claim-states/part-admission-states.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  PA_PAY_IN_INSTALMENTS = 'pa-pay-in-instalments',
```
Comment (robertparkinson) [inline src/main/app/claims/models/claim-states/part-admission-states.ts]: same as above

File: src/main/app/claims/models/claim-states/part-admission-states.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  PA_PAY_BY_SPECIFIED_DATE_REFERRED_TO_JUDGE = 'pa-pay-by-specified-date-referred-to-judge',
+  PA_PAY_IN_INSTALMENTS_REFERRED_TO_JUDGE = 'pa-pay-in-instalments-referred-to-judge',
+
+  PA_CCJ_PAY_BY_SET_DATE_BY_ADMISSION = 'pa-ccj-pay-by-set-date-by-admission',
```
Comment (robertparkinson) [inline src/main/app/claims/models/claim-states/part-admission-states.ts]: this doesn't seem to be used either?

File: src/main/app/claims/models/claim-states/states-paid-states.ts (hunk 1 above)
Comment (andrewwa-kainos) [reply src/main/app/claims/models/claim-states/states-paid-states.ts]: So is this just for part admit states paid? What about the substates under that?

Comment (BenDoherty0103) [reply src/main/app/claims/models/claim-states/states-paid-states.ts]: Correct. I will be removing this states paid state from this story as it (and its substates) are being addressed in a separate story (https://tools.hmcts.net/jira/browse/ROC-5267, I'll attach a screenshot below of my WIP for that story), this state was already there when I started work on this story :)

Screen Shot 2019-06-27 at 14 17 18

**Comment by robertparkinson** [comment]
took out route tests into their own PR: https://github.com/hmcts/cmc-citizen-frontend/pull/1960
the rest of this work will have to be looked at after we refactor to the state flow model

**Pull request closed without merging**